arelle/plugin/validate/ESEF_2022/Util.py

```diff
@@ -55,41 +60,98 @@ def isInEsefTaxonomy(val: ValidateXbrl, modelObject: ModelObject | None) -> bool
     False: ("gif", "jpeg", "png") # mime types: jpg is not a valid mime type
     }
 # check image contents against mime/file ext and for Steganography
-def checkImageContents(modelXbrl: ModelXbrl, imgElt: ModelObject, imgType: str, isFile: bool, data: bytes) -> None:
+
+def validateImage(baseUrl:Optional[str], image: str, modelXbrl: ModelXbrl, val:ValidateXbrl, elt:_Element, evaluatedMsg:str, contentOtherThanXHTMLGuidance:str) -> None:
+    """
+    image: either an url or base64 in data:image style
+    """
+    minExternalRessourceSize = val.authParam["minExternalResourceSizekB"]
+    if minExternalRessourceSize != -1:
+        # transform kb to b
+        minExternalRessourceSize = minExternalRessourceSize * 1024
+    if scheme(image) in ("http", "https", "ftp"):
+        modelXbrl.error("ESEF.4.1.6.xHTMLDocumentContainsExternalReferences" if val.unconsolidated
+                        else "ESEF.3.5.1.inlineXbrlDocumentContainsExternalReferences",
+                        _("Inline XBRL instance documents MUST NOT contain any reference pointing to resources outside the reporting package: %(element)s"),
+                        modelObject=elt, element=elt.tag, evaluatedMsg=evaluatedMsg,
+                        messageCodes=("ESEF.3.5.1.inlineXbrlDocumentContainsExternalReferences",
```

Comment:
nit: I'd flip the elif and else cases here to avoid the negative assertion:
`elif image.startswith("data:image"):` instead of `elif not image.startswith("data:image"):`

Reply:
The suggested change has been done!